File: driven-development/1. DDD START! 도메인 주도 설계 구현과 핵심 개념 익히기/src/main/java/com/myshop/common/jpa/converter/MoneyConverter.java
package com.myshop.common.jpa.converter;

import com.myshop.common.model.Money;

import javax.persistence.AttributeConverter;

public class MoneyConverter implements AttributeConverter<Money, Integer> {

    @Override
    public Integer convertToDatabaseColumn(Money money) {
        return money == null ? null : money.getValue();
    }

    @Override
    public Money convertToEntityAttribute(Integer dbData) {
        return dbData == null ? null : new Money(dbData);
    }
}

File: spring/Spring-JWT/core-jpa/src/main/kotlin/com/github/keunwon/corejpa/convert/MoneyConverter.kt
package com.github.keunwon.corejpa.convert

import com.github.keunwon.core.generic.money.Money
import java.math.BigDecimal
import javax.persistence.AttributeConverter
import javax.persistence.Converter

@Converter(autoApply = true)
class MoneyConverter : AttributeConverter<Money, BigDecimal> {
    override fun convertToDatabaseColumn(money: Money): BigDecimal = money.amount

    override fun convertToEntityAttribute(dbData: BigDecimal): Money = Money.wons(dbData)
}
